kivy/uix/scrollview.py

```diff
@@ -763,8 +763,8 @@ def on_scroll_move(self, touch):
         if ud['mode'] == 'unknown':
             ud['dx'] += abs(touch.dx)
             ud['dy'] += abs(touch.dy)
-            if ((ud['dx'] > self.scroll_distance) or
-                    (ud['dy'] > self.scroll_distance)):
+            if ((ud['dx'] > self.scroll_distance and self.do_scroll_x) or
+                    (ud['dy'] > self.scroll_distance and self.do_scroll_y)):
```

Comment:
do_scroll_x = True
do_scroll_y = False
scroll_distance > ud['dx']
scroll_distance > ud['dy']

    if ((True and True) or (True and False)):  # True, k, let's go further
        if not True and not False:                    # False, oops.

The bottom part won't be executed unless both scrollings are enabled. Was it intender or am I the only one confused here?